python/ray/train/tests/test_mosaic_trainer.py

```diff
@@ -0,0 +1,138 @@
+import pytest
+
+import torch
+import torch.utils.data
+
+import torchvision
+from torchvision import transforms, datasets
+
+from torchmetrics.classification.accuracy import Accuracy
+from composer.core.evaluator import Evaluator
+from composer.models.tasks import ComposerClassifier
+import composer.optim
+from composer.loggers import InMemoryLogger
+from composer.algorithms import LabelSmoothing
+
+import ray
+from ray.air.config import ScalingConfig
+import ray.train as train
+from ray.air import session
+from ray.train.mosaic import MosaicTrainer
+
+
+mean = (0.507, 0.487, 0.441)
+std = (0.267, 0.256, 0.276)
+cifar10_transforms = transforms.Compose(
+    [transforms.ToTensor(), transforms.Normalize(mean, std)]
+)
+
+data_directory = "s3://air-example-data/cifar-10-python.tar.gz"
+train_dataset = torch.utils.data.Subset(
+    datasets.CIFAR10(
+        data_directory, train=True, download=True, transform=cifar10_transforms
+    ),
+    list(range(64)),
+)
+test_dataset = torch.utils.data.Subset(
+    datasets.CIFAR10(
+        data_directory, train=False, download=True, transform=cifar10_transforms
+    ),
+    list(range(64)),
+)
```

Comment:
can we move all the dataset creation to inside the `trainer_init_per_worker`? That way we don't have to serialize the entire dataset and send it to each worker.

Reply:
Done!